### PR TITLE
Refactor/df 1891 event sender receiver

### PR DESCRIPTION
**Pull request by cjpsycos**

### CODE REVIEW

File: infrastructure/src/main/java/ch/admin/seco/alv/service/competencecatalog/infrastructure/messagebroker/receiver/EventReceiverGateway.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        this.competenceElementApplicationService = competenceElementApplicationService;
+        this.competenceSetApplicationService = competenceSetApplicationService;
+        this.chFicheApplicationService = chFicheApplicationService;
+    }
+
+    @StreamListener(target = COMPETENCECATALOG_EVENT_CHANNEL, condition = COMPETENCE_ELEMENT_PUBLISHED_CONDITION)
+    @LoginAsTechnicalUser
+    public void handlePublishedEvent(CompetenceElementPublishedEventDto eventDto) {
+        if (competenceElementApplicationService.findById(eventDto.getCompetenceElement().getId()) == null) {
+            CreateCompetenceElementDto createCompetenceElementDto = new CreateCompetenceElementDto();
```
Comment (emina-str) [inline infrastructure/src/main/java/ch/admin/seco/alv/service/competencecatalog/infrastructure/messagebroker/receiver/EventReceiverGateway.java]: @cjpsycos  we already use Builder setters, so why not just concatenate multiple setters? (just a suggestion)
```
CreateCompetenceElementDto createCompetenceElementDto = new CreateCompetenceElementDto()
            .setType(eventDto.getCompetenceElement().getType())
            .setDescription(eventDto.getCompetenceElement().getDescription());
```

File: infrastructure/src/main/java/ch/admin/seco/alv/service/competencecatalog/infrastructure/messagebroker/receiver/EventReceiverGateway.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import ch.admin.seco.alv.service.competencecatalog.application.security.LoginAsTechnicalUser;
+import ch.admin.seco.alv.service.competencecatalog.domain.chfiche.ChFicheId;
+import ch.admin.seco.alv.service.competencecatalog.domain.competenceelement.CompetenceElementId;
+import ch.admin.seco.alv.service.competencecatalog.domain.competenceset.CompetenceSetId;
+import ch.admin.seco.alv.service.competencecatalog.infrastructure.messagebroker.*;
+import org.springframework.cloud.stream.annotation.StreamListener;
+
+import static ch.admin.seco.alv.service.competencecatalog.infrastructure.messagebroker.MessageBrokerChannels.*;
+
+public class EventReceiverGateway {
```
Comment (emina-str) [inline infrastructure/src/main/java/ch/admin/seco/alv/service/competencecatalog/infrastructure/messagebroker/receiver/EventReceiverGateway.java]: Question, we don't need her `@Component` annotation?

Comment (emina-str) [reply infrastructure/src/main/java/ch/admin/seco/alv/service/competencecatalog/infrastructure/messagebroker/receiver/EventReceiverGateway.java]: I saw `@Bean` ...

File: infrastructure/src/main/java/ch/admin/seco/alv/service/competencecatalog/infrastructure/messagebroker/receiver/EventReceiverGateway.java (hunk 1 above)
Comment (cjpsycos) [reply infrastructure/src/main/java/ch/admin/seco/alv/service/competencecatalog/infrastructure/messagebroker/receiver/EventReceiverGateway.java]: true. will be done

Comment (dvoracek-martin) [reply infrastructure/src/main/java/ch/admin/seco/alv/service/competencecatalog/infrastructure/messagebroker/receiver/EventReceiverGateway.java]: done

**Pull request merged**